### ISSUE TITLE
No video texture for Unity3d 5.2.1

### ISSUE THREAD

**Issue by thekalla** [artoolkit/arunity5#20]
Pressing play button in Unity Editor 5.2.0f3 displays a pink texture for SimpleScene project on MacOS 10.11

**Comment by weihuan** [comment]
No video texture for Unity3d 5.2.1 on windows 7 64 bit. unity 5.2.0 the same problem, help!

**Comment by wallywyyoung** [comment]
Hey Guys, definitely confirmed. Will look into it shortly. Has anyone had success getting around this?

**Comment by JustinFincher** [comment]
```Shader warning in 'VideoPlaneNoLight': Shader is not supported. Are you creating a fixed function shader using Material(string) constructor?```  
IN 5.2 UNITY  Can not create fixed function shaders using "new Material(string)" anymore. Shaders must come from assets or be fully precompiled in the editor. SEE:[http://unity3d.com/cn/unity/beta/unity5.2.0f1](http://unity3d.com/cn/unity/beta/unity5.2.0f1)  
### Current Workaround:  
just create a shader and replace the content with code:
```
Shader "VideoPlaneNoLight" 
{
  Properties 
  {
    _Color ("Main Color", Color) = (1,1,1,1)
    _MainTex ("Base (RGB)", 2D) = "white" { }
  }
  SubShader 
  {
    Pass 
    {
      Material 
      {
        Diffuse [_Color]
      }
      Lighting Off
      ZWrite Off
      Blend SrcAlpha OneMinusSrcAlpha
      SeparateSpecular Off
      SetTexture [_MainTex] 
      { 
        constantColor [_Color]
        Combine texture * constant, texture * constant
      }
    }
  }
}
```
And in `ARController.cs`: line 1283
```
vbm = new Material(shaderSource); //ARToolKit5-Unity.Properties.Resources.VideoPlaneShader;
```
replace it with:
```
vbm = new Material(Shader.Find("VideoPlaneNoLight"));
```

Then camera texture shows.  

### but I still get another error when running:  
```
NullReferenceException: Object reference not set to an instance of an object
UnityEditor.DockArea.OnGUI () (at /Users/builduser/buildslave/unity/build/Editor/Mono/GUI/DockArea.cs:264)
ARNativePlugin:arwStartRunningB(String, Byte[], Int32, Single, Single)
PluginFunctions:arwStartRunningB(String, Byte[], Int32, Single, Single) (at Assets/ARToolKit5-Unity/Scripts/PluginFunctions.cs:108)
ARController:StartAR() (at Assets/ARToolKit5-Unity/Scripts/ARController.cs:563)
ARController:Start() (at Assets/ARToolKit5-Unity/Scripts/ARController.cs:359)
```
Does anyone knows about this error ? I'm on Unity 5.2.2f1

**Comment by JustinFincher** [comment]
Oh seems [22](https://github.com/artoolkit/arunity5/pull/22/files) already did this... but why I still get this problem using .unitypackage downloaded from [http://artoolkit.org/](http://artoolkit.org/) ?

**Comment by wallywyyoung** [comment]
We haven't updated our package release on artoolkit.org yet, we will be soon. Thank you!

**Comment by wallywyyoung** [comment]
Should be fixed in 5.3.1 release. Please double check in the newest version. Thanks!
